test/unit/metrics/test_metricframe_streaming.py

```diff
@@ -148,3 +154,72 @@ def test_exception_on_no_data():
     with pytest.raises(ValueError) as excinfo:
         _ = target.control_levels
     assert "No data to process" in str(excinfo.value)
+
+
+@pytest.mark.parametrize("transform_y_p", conversions_for_1d)
+@pytest.mark.parametrize("transform_y_t", conversions_for_1d)
+@pytest.mark.parametrize("num_batches", [1, 3, 5])
+def test_cf_argument_mismatch(transform_y_t, transform_y_p, num_batches):
+    batch_generator = batchify(num_batches, y_t, y_p, g_4)
+
+    # We initialize without CF
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 streaming=True)
+    # We supply CF in the call, it should fail.
+    with pytest.raises(ValueError):
+        y_t_batches, y_p_batches, g_4_batches = next(batch_generator)
+        g_f = pd.DataFrame(data=g_4_batches, columns=['My feature'])
+        target.add_data(transform_y_t(y_t_batches),
+                        transform_y_p(y_p_batches),
+                        sensitive_features=g_f,
+                        control_features=g_f)
+
+    batch_generator = batchify(num_batches, y_t, y_p, g_4)
+    # We initialize with CF
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 control_features=[], streaming=True)
+    # We supply CF in the call.
+    y_t_batches, y_p_batches, g_4_batches = next(batch_generator)
+    g_f = pd.DataFrame(data=g_4_batches, columns=['My feature'])
+    target.add_data(transform_y_t(y_t_batches),
+                    transform_y_p(y_p_batches),
+                    sensitive_features=g_f,
+                    control_features=g_f)
+
+
+@pytest.mark.parametrize("transform_y_p", conversions_for_1d)
+@pytest.mark.parametrize("transform_y_t", conversions_for_1d)
+def test_sp_argument_mismatch(transform_y_t, transform_y_p):
+    num_batches = 2
+    sample_weights = np.random.rand(len(y_t))
+    batch_generator = batchify(num_batches, y_t, y_p, g_4, sample_weights)
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 sample_params=None, streaming=True)
+
+    y_t_batches, y_p_batches, g_4_batches, sw_batches = next(batch_generator)
+    g_f = pd.DataFrame(data=g_4_batches, columns=['My feature'])
+    target.add_data(transform_y_t(y_t_batches),
+                    transform_y_p(y_p_batches),
+                    sensitive_features=g_f,
+                    sample_params={'sample_weight': sw_batches})
+
+    # We forget to supply it the next time, it should raise.
+    with pytest.raises(ValueError):
+        target.add_data(transform_y_t(y_t_batches),
+                        transform_y_p(y_p_batches),
+                        sensitive_features=g_f)
+
+    # Version with sample_weights supplied at the start.
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 sample_params={'sample_weight': sample_weights},
```

Comment:
Shouldn't this be `sample_params={'sample_weight': [] }` here - the sample weights would need to be streamed as well?